src/cmap/errors.ts

```diff
@@ -27,10 +27,11 @@ export class PoolClearedError extends MongoNetworkError {
   /** The address of the connection pool */
   address: string;
 
-  constructor(pool: ConnectionPool) {
-    super(
-      `Connection pool for ${pool.address} was cleared because another operation failed with: "${pool.serverError?.message}"`
-    );
+  constructor(pool: ConnectionPool, message?: string) {
```

Comment:
PoolClearedError is not exported (but it's not marked internal), so it should be okay to modify the constructor.